### PR TITLE
bug/webapp 3174

### PR DESCRIPTION
**Pull request by herrmannplatz**
- simplifies `_handle_event` and adds proper check what to store
- combines ignored and deprecated events
- deletes all 'member-update' events in db

### CODE REVIEW

File: app/script/event/EventRepository.coffee [hunk 1]
```diff
@@ -382,17 +368,10 @@ class z.event.EventRepository
...
-        .then ->
-          proceed()
         .catch (error) =>
-          if error.message is z.event.EventError::TYPE.DEPRECATED_SCHEMA
-            @logger.log @logger.levels.WARN, "Ignored notification '#{notification.id}' from '#{source}': #{error.message}", error
-            proceed()
-          else
-            @logger.log @logger.levels.ERROR, "Failed to handle notification '#{notification.id}' from '#{source}': #{error.message}", error
-            reject error
+          @logger.log @logger.levels.ERROR, "Failed to handle notification '#{notification.id}' from '#{source}': #{error.message}", error
```
Comment (gregor) [inline app/script/event/EventRepository.coffee]: Remove source as you no longer have access to it

File: app/script/event/EventTypeHandling.coffee [hunk 2]
```diff
@@ -31,15 +31,13 @@ z.event.EventTypeHandling =
     z.event.Backend.CONVERSATION.OTR_ASSET_ADD
     z.event.Backend.CONVERSATION.OTR_MESSAGE_ADD
   ]
-  DEPRECATED: [
+  IGNORE: [
+    z.event.Backend.CONVERSATION.TYPING
```
Comment (gregor) [inline app/script/event/EventTypeHandling.coffee]: Alphabet

File: app/script/event/EventRepository.coffee [hunk 3]
```diff
@@ -340,7 +340,7 @@ class z.event.EventRepository
     .then (saved_event) =>
       @_distribute_event saved_event
       return saved_event
-    .catch (error) =>
+    .catch (error) ->
       if error.type is z.cryptography.CryptographyError::TYPE.PREVIOUSLY_STORED
         return true
       else
```
Comment (lipis) [inline app/script/event/EventRepository.coffee]: Not part of this PR but since we are here.. no need for `else` since the if `returns`

```coffeescript
if error.type is z.cryptography.CryptographyError::TYPE.PREVIOUSLY_STORED
  return true
throw error
```

File: app/script/event/EventRepository.coffee [hunk 4]
```diff
@@ -319,46 +319,32 @@ class z.event.EventRepository
...
-        @_distribute_event record if record
-        resolve record
-      .catch (error) =>
-        if error.type is z.cryptography.CryptographyError::TYPE.PREVIOUSLY_STORED
-          resolve true
-        else
-          @logger.log @logger.levels.ERROR,
-            "Failed to handle '#{event.type}' event '#{event.id or 'no ID'}' from '#{source}': '#{error.message}'", event
-          reject error
+        throw error
```
Comment (lipis) [inline app/script/event/EventRepository.coffee]: no need for `else` since the if `returns`

``` coffeescript
if error.type is z.cryptography.CryptographyError::TYPE.PREVIOUSLY_STORED
  return true
throw error
```

File: app/script/event/EventRepository.coffee [hunk 5]
```diff
@@ -320,46 +320,31 @@ class z.event.EventRepository
...
+    Promise.resolve()
+    .then =>
+      if event.type in z.event.EventTypeHandling.DECRYPT
+        return @cryptography_repository.decrypt_event(event)
+        .then (generic_message) =>
+          @cryptography_repository.cryptography_mapper.map_generic_message generic_message, event
+      return event
+    .then (mapped_event) =>
+      if mapped_event.type in z.event.EventTypeHandling.STORE
+        return @cryptography_repository.save_unencrypted_event mapped_event
```
Comment (gregor) [inline app/script/event/EventRepository.coffee:339]: Should be `save_event` now
